### PR TITLE
[WIP] feat: deployment pusher impelementation

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #

### CODE REVIEW

File: packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx [hunk 1]
```diff
@@ -47,44 +50,148 @@ const buildStatusToReadable = (status: string): string =>
...
-        <Subtitle>Location</Subtitle>
-        <BodyText>{pipeline.subDomain ? `https://${pipeline.subDomain}.dev.paas.reapit.cloud` : ''}</BodyText>
-      </InputWrap>
-    </ColSplit>
-  </Grid>
-)
-
-export const PipelineDeploymentInfo = ({ pipeline }: { pipeline: PipelineModelInterface }) => {
-  // TODO make some fetching func to get pipeline deployments and list them below
+const PipelineInfo = ({ pipeline }: { pipeline: PipelineModelInterface }) => {
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx:53]: Can you add a type here please ie `FC<SomeProps`

File: packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx [hunk 2]
```diff
@@ -47,44 +50,148 @@ const buildStatusToReadable = (status: string): string =>
...
+        <InputWrap>
+          <Subtitle>Location</Subtitle>
+          <BodyText>{pipeline.subDomain ? `https://${pipeline.subDomain}.dev.paas.reapit.cloud` : ''}</BodyText>
+        </InputWrap>
+      </ColSplit>
+    </Grid>
+  )
+}
+
+const PipelineDeploymentTable = ({
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx:91]: As above, can you add a type. Also, can you break this component into a separate file please

File: packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx (hunk 1 above)
Comment (willmcvay) [reply packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx:53]: Also, needs a test

File: packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx [hunk 3]
```diff
@@ -47,44 +50,148 @@ const buildStatusToReadable = (status: string): string =>
...
+          <Subtitle>Build Command</Subtitle>
+          <BodyText>{pipeline.buildCommand}</BodyText>
+        </InputWrap>
+        <InputWrap>
+          <Subtitle>Tests</Subtitle>
+          <BodyText>{pipeline.testCommand}</BodyText>
+        </InputWrap>
+        <InputWrap>
+          <Subtitle>Location</Subtitle>
+          <BodyText>{pipeline.subDomain ? `https://${pipeline.subDomain}.dev.paas.reapit.cloud` : ''}</BodyText>
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx:84]: What happens in prod? The dev.paas bit should come from the config so we can switch it for prod.paas

File: packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx [hunk 4]
```diff
@@ -47,44 +50,148 @@ const buildStatusToReadable = (status: string): string =>
...
+  newRunner,
+}: {
+  pipeline: PipelineModelInterface
+  initialDeployments: null | { items: PipelineRunnerModelInterface[] }
+  loading: boolean
+  channel: any
+  newRunner: PipelineRunnerModelInterface | undefined
+}) => {
+  const [pagination, setPagination] = useState<{ items: PipelineRunnerModelInterface[] } | null>(initialDeployments)
+  useEffect(() => {
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx:105]: Can you break the useEffect functions into handlers and add tests please

File: packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx [hunk 5]
```diff
@@ -47,44 +50,148 @@ const buildStatusToReadable = (status: string): string =>
...
+            <TableCell>{deployment.currentlyDeployed ? 'Deployed' : ''}</TableCell>
+            <TableCell></TableCell>
+          </TableRow>
+        ))
+      ) : null}
+    </Table>
+  )
+}
+
+export const PipelineDeploymentInfo = ({ pipeline, channel }: { pipeline: PipelineModelInterface; channel: any }) => {
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/v8/pipeline/deployment-info.tsx:194]: As above, separate component, add a type and tests

File: packages/developer-portal/src/components/pages/v8/pipeline/index.tsx [hunk 6]
```diff
@@ -1,12 +1,49 @@
...
 import { PipelineDeploymentInfo } from './deployment-info'
 import { reapitConnectBrowserSession } from '../../../../core/connect-session'
 import { GetActionNames, getActions } from '@reapit/utils-common'
 import { BodyText, FlexContainer, Loader } from '@reapit/elements'
-import { useReapitConnect } from '@reapit/connect-session'
+import { ReapitConnectSession, useReapitConnect } from '@reapit/connect-session'
+import { PusherProvider, useEvent, useChannel } from '@harelpls/use-pusher'
+import { URLS, COGNITO_HEADERS } from '../../../../constants/api'
+
+const PipelineWrapper = ({
```
Comment (willmcvay) [inline packages/developer-portal/src/components/pages/v8/pipeline/index.tsx:13]: And again for these two components, can you break out the logic, break into separate components, add tests and props types please

File: packages/utils-react/src/use-reapit-data/use-reapit-update.ts [hunk 7]
```diff
@@ -76,18 +79,22 @@ export const send =
 
     if (!getHeaders) throw new Error('Missing valid Reapit Connect Session')
 
+    console.log('starting')
```
Comment (willmcvay) [inline packages/utils-react/src/use-reapit-data/use-reapit-update.ts:82]: Do you need this console?

File: packages/utils-react/src/use-reapit-data/use-reapit-update.ts [hunk 8]
```diff
@@ -76,18 +79,22 @@ export const send =
...
+      console.log('fetching')
+
       const response = await fetch(url, {
         headers: getHeaders,
         method,
         body: JSON.stringify(params),
       })
 
-      if (!returnUpdatedModel) await setLoading(false)
+      if (!returnUpdatedModel && !returnUpdatedModelWithLocation) await setLoading(false)
```
Comment (willmcvay) [inline packages/utils-react/src/use-reapit-data/use-reapit-update.ts:95]: There really is a lot of logic in here - is there a way to break it out into something more maintainable / testable? I am just thinking it's something we will use all over the place and feels a bit fragile / complex

**Review by willmcvay — CHANGES_REQUESTED** [review]

**Review by joshbalfour — APPROVED** [review]

**Comment by bashleigh** [comment]
Needs live urls though (!)

**Review by willmcvay — APPROVED** [review]
Changes look better, can you rebase and get it merged in please :-)

**Pull request merged**